.github/workflows/icu4c.yml

```
@@ -67,6 +67,8 @@ jobs:
 
       - name: ICU4C with gcc
         env:
+          CXXFLAGS: -Wextra -Werror -Wno-return-local-addr
```

```suggestion
          CXXFLAGS: -Wextra -Werror -Wno-error=return-local-addr
```

I recommend not hiding this warning entirely, but instead just not treat it as an error.